### PR TITLE
.github: Migrate pytorch_linux_bionic_py_3_6_clang9 to GHA

### PR DESCRIPTION
**Pull request by seemethere**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64320
* __->__ #64218

Relies on https://github.com/fairinternal/pytorch-gha-infra/pull/11

Includes both ~xla~ / noarch jobs as well (xla moved to its own PR)

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

cc @ezyang @seemethere @malfet @walterddr @lg20987 @pytorch/pytorch-dev-infra @bdhirsh

Differential Revision: [D30651516](https://our.internmc.facebook.com/intern/diff/D30651516)

### CODE REVIEW

**Review by malfet — APPROVED** [review]

**Comment by malfet** [comment]
Please delete CirclecCI workflow as part of this PR as well

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30651516).

File: .jenkins/pytorch/common_utils.sh [hunk 1]
```diff
@@ -99,5 +99,7 @@ function checkout_install_torchvision() {
 }
 
 function clone_pytorch_xla() {
-  git clone --recursive https://github.com/pytorch/xla.git
+  if [[ ! -d './xla' ]]; then
```
Comment (seemethere) [inline .jenkins/pytorch/common_utils.sh:102]: if directory exists this is a noop

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30651516).

File: .jenkins/pytorch/test.sh [hunk 2]
```diff
@@ -508,6 +519,7 @@ if [[ "${BUILD_ENVIRONMENT}" == *backward* ]]; then
   # Do NOT add tests after bc check tests, see its comment.
 elif [[ "${BUILD_ENVIRONMENT}" == *xla* || "${JOB_BASE_NAME}" == *xla* ]]; then
   install_torchvision
+  build_xla
```
Comment (seemethere) [inline .jenkins/pytorch/test.sh:522]: No idea why were previously dedicating an entire job just to building xla when the base pytorch remained the same.

File: .jenkins/pytorch/test.sh [hunk 3]
```diff
@@ -407,9 +407,21 @@ test_torch_function_benchmark() {
   assert_git_not_dirty
 }
 
+build_xla() {
+  clone_pytorch_xla
+  # shellcheck disable=SC1091
+  source "xla/.circleci/common.sh"
+  apply_patches
+  install_deps_pytorch_xla xla
+  set_torch_cmake_prefix_path
```
Comment (seemethere) [inline .jenkins/pytorch/test.sh:416]: Surprisingly `xla/.circleci/common.sh` also decides to muck around with `CMAKE_PREFIX_PATH` so we need to set it after we source `xla/.circleci/common.sh`

Comment (seemethere) [reply .jenkins/pytorch/test.sh:416]: Submitted https://github.com/pytorch/xla/pull/3106 to fix

**Comment by seemethere** [comment]
This PR is currently blocked on https://github.com/pytorch/xla/issues/3107

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30651516).

**Comment by seemethere** [comment]
I've gone ahead and removed the xla jobs from this PR and shuffled them to a dependent PR

**Review by malfet — APPROVED** [review]

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30651516).

**Review by janeyx99 — APPROVED** [review]

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30651516).

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30651516).

**Pull request closed without merging**